zhaquirks/tuya/builder/__init__.py

```diff
@@ -443,18 +443,31 @@ def tuya_dp(
         """Add Tuya DP Converter."""
         self.tuya_dp_to_attribute.update(
             {
-                dp_id: DPToAttributeMapping(
-                    ep_attribute,
-                    attribute_name,
-                    converter=converter,
-                    dp_converter=dp_converter,
-                    endpoint_id=endpoint_id,
-                )
+                dp_id: [
+                    DPToAttributeMapping(
+                        ep_attribute,
+                        attribute_name,
+                        converter=converter,
+                        dp_converter=dp_converter,
+                        endpoint_id=endpoint_id,
+                    )
+                ]
             }
         )
         self.tuya_data_point_handlers.update({dp_id: dp_handler})
         return self
 
+    def tuya_dp_multi(
+        self,
+        dp_id: int,
+        attribute_mapping: list[DPToAttributeMapping],
+        dp_handler: str = "_dp_2_attr_update",
+    ) -> QuirkBuilder:  # fmt: skip
+        """Add Tuya DP Converter that maps to multiple attributes."""
+        self.tuya_dp_to_attribute.update({dp_id: attribute_mapping})
+        self.tuya_data_point_handlers.update({dp_id: dp_handler})
+        return self
```

Comment:
Hmm, is there any benefit to also having this method instead of just calling `tuya_dp` twice for the same endpoint?

If so, we may want to add a test for this, though I'd need to check if `tuya_dp` is tested explicitly (or just by one of the other Tuya methods).

Reply:
Part of the reason was that calling `tuya_dp` multiple times would allow passing a different `dp_handler`, which will not work properly. It would still handle multiple `DPToAttributeMapping`, but just overwrite the `dp_handler`.
Having a separate method prevents that and makes it clearer to who uses it. I can drop it if you think it would be better that way.

Reply:
Oh, I just noticed that I missed the tests when opening this PR from the original one: https://github.com/zigpy/zha-device-handlers/pull/3644/files#diff-d01eb6338c5bfa4635fb433e6a8fc5b9cf3de58e784218f8930c5ccaedd927f5
:man_facepalming:

Reply:
Ok, one more thing regarding this. Can we either block subsequent calls to `tuya_dp` if the `dp_id` is already in `tuya_dp_to_attribute` or just have it update the existing list of `DPToAttributeMapping` per id?
For the latter, both `tuya_dp_multi` and subsequent calls to `tuya_dp` would work as expected then.

Reply:
Added that in a new PR https://github.com/zigpy/zha-device-handlers/pull/3827 since it required changing 2 quirks that were invalid.

Reply:
Updated the `tuya_dp` method to call `tuya_dp_multi` since the code is mostly the same internally.